sql/core/src/test/scala/org/apache/spark/sql/execution/command/AlterTableAddPartitionSuiteBase.scala

```diff
@@ -181,4 +181,37 @@ trait AlterTableAddPartitionSuiteBase extends QueryTest with DDLCommandTestUtils
       checkPartitions(t, Map("id" -> "1"), Map("id" -> "2"))
     }
   }
+
+  test("SPARK-34304: adding partitions to views is not allowed") {
+    withNamespaceAndTable("ns", "tbl") { t =>
+      sql(s"CREATE TABLE $t (id INT, part INT) $defaultUsing PARTITIONED BY (part)")
+      def checkViewAltering(createViewCmd: String, alterCmd: String): Unit = {
+        sql(createViewCmd)
+        val errMsg = intercept[AnalysisException] {
+          sql(alterCmd)
+        }.getMessage
+        assert(errMsg.contains("'ALTER TABLE ... ADD PARTITION ...' expects a table"))
+        checkPartitions(t) // no partitions
+      }
+
+      withView("v0") {
```

Comment:
shall we put the test  in `SQLViewTestSuite`? Then it can be tested with 3 view types automatically.

Reply:
Same question to other tests in this PR.

Reply:
> ... it can be tested with 3 view types automatically

1. I think I can rewrite the test and check all 3 view type automatically as well like `withAllViews(view) { ... }`
2. The test here runs on 3 catalogs, and also automatically as you know.

From my understanding, `SQLViewTestSuite` should focus on view functionality in general but unified DS tests focus is on specific commands. I would prefer to not pollute `SQLViewTestSuite` by specific DDL commands.

Another benefit to have the test here is we can check entire command behavior just by running;
`build/sbt "test:testOnly *AlterTableAddPartitionSuite"`

Reply:
But this view check is done by the analyzer, and it doesn't matter what the underlying catalog is. Is it a waste to test this with 3 different catalogs?